### PR TITLE
Windows qttestdriver

### LINKED ISSUE #292: QTestDriver Implementation not working on Windows
### 👓 What did you see?

Using the provided QTestDriver Implementation for windows gives  errors

` Unable to open file for logging: %TEMP%/test_module_runner.XNlhxg`

The filename `test_module_runner` comes from the executable name.

### ✅ What did you expect to see?

The file should be created and used for QTest logging.

### 📦 Which tool/library version are you using?

cucumber-cpp v0.7.0

### 🔬 How could we reproduce it?

run QTestDriver test

### 📚 Any additional context?

This fails cause the QTemporary File created in the QTestDriver implementation is not going ouf of scope and therefore exists and is opened by cucumber-cpp. qTestExec could not write to this file and give the above error from Qt Internals.

The Temporary file name should created another way or the QTemporary File must go out of scope to let windows access the file resource.

### PR DESCRIPTION
**Pull request by ursfassler**
## Summary

This is the code part of PR #293. As suggested there, the Windows test for CI will be handled separately.

## Details

Fixes #292.

## Motivation and Context

See #293.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] It is my own work, its copyright is implicitly assigned to the project and no substantial part of it has been copied from other sources (including [Stack Overflow](https://stackoverflow.com/)). In rare occasions this is acceptable, like in CMake modules where the original copyright information should be kept.
- [x] I'm using the same code standards as the existing code (indentation, spacing, variable naming, ...).
- [ ] I've added tests for my code.
- [x] I have verified whether my change requires changes to the documentation
- [x] My change either requires no documentation change or I've updated the documentation accordingly.
- [ ] My branch has been rebased to main, keeping only relevant commits.

### CODE REVIEW

**Comment by ursfassler** [comment]
@kreuzberger I took the production code changes from your PR #293 and did a small cleanup. The goal is to have this changes merged; we will see what we do about the windows tests later.

I would appreciate it if you test this changes on windows. If everything is ok, I'll squash and merge this PR.

**Comment by kreuzberger** [comment]
> @kreuzberger I took the production code changes from your PR #293 and did a small cleanup. The goal is to have this changes merged; we will see what we do about the windows tests later.
> 
> I would appreciate it if you test this changes on windows. If everything is ok, I'll squash and merge this PR.

I have build and run the tests for config release for this branch.
TLDR: The tests were successfull (exluded boost tests)

The first test run didnt work cause the test environment is NOT properly set for the QtTests (see the other changes in my PRS).
AFTER adding Qt binary dir to PATH environment variable  the tests were successfull.
